### PR TITLE
refactor: Replace Status with BTStatus

### PR DESCRIPTION
**Pull request by ThePat02**
Replaced Status with BTStatus. This was done by search and replace and needs to be tested!

### CODE REVIEW

**Comment by ThePat02** [comment]
> Blackboard to BTBlackboard

No, since this can be used with FSM too, so maybe `BehaviourBlackboard`

**Comment by SirPigeonz** [comment]
Lol I though BT stands for Behavior Toolkit haha maybe we should think about our own prefix for all classes to avoid collusions like `Bt` ?

**Comment by SirPigeonz** [comment]
GDScript needs namespaces...

**Comment by SirPigeonz** [comment]
> I think just `Behaviour` should do.

Yeah it's fine.

> I really want to keep things like `FiniteStateMachine` without any additional Pre or suffixes.

Yeah it's a paint in ass. I'm just concerned if use would want to use other addon with conflicting name (it's generic name it's high chance for it). GDScript needs namespaces for addons lol xD

**Review by SirPigeonz — CHANGES_REQUESTED** [review]

File: addons/behaviour_toolkit/behaviour_tree/bt_root.gd [hunk 1]
```diff
@@ -25,7 +25,7 @@ enum ProcessType {
 
 
 var active: bool = false
-var current_status: BTBehaviour.Status
+var current_BTStatus: BTBehaviour.BTStatus
```
[the remark this answers is not in the document]
Comment (ThePat02) [reply addons/behaviour_toolkit/behaviour_tree/bt_root.gd:28]: > While doing refactoring like that I recommend to use the refactoring tool as a jump-to tool and approve each change manually. :)

Yes! I just tried replacing all of it for the ease of things before diving deeper into the code.

File: addons/behaviour_toolkit/behaviour_tree/leaves/leaf_event.gd
Comment (ThePat02) [inline addons/behaviour_toolkit/behaviour_tree/leaves/leaf_event.gd:1]: Same with `return_BTStatus`

File: addons/behaviour_toolkit/finite_state_machine/fsm.gd
Comment (ThePat02) [inline addons/behaviour_toolkit/finite_state_machine/fsm.gd:1]: Same issue

File: addons/behaviour_toolkit/finite_state_machine/fsm_state_integrated_bt.gd
Comment (ThePat02) [inline addons/behaviour_toolkit/finite_state_machine/fsm_state_integrated_bt.gd:1]: Same Issue

**Comment by ThePat02** [comment]
Needs testing, as I fixed this in codespaces.

**Comment by SirPigeonz** [comment]
I've checked the code and looks good! At least what it's worth, my atention span today is of a golden retriever xD

Will take this PR on test ride today night :)

File: addons/behaviour_toolkit/behaviour_tree/composites/bt_integrated_fsm.gd [hunk 2]
```diff
@@ -5,14 +5,14 @@ class_name BTIntegratedFSM extends BTComposite
...
 
 
-func tick(_actor: Node, _blackboard: Blackboard) -> Status:
+func tick(_actor: Node, _blackboard: Blackboard) -> BTStatus:
 	if state_machine.active == false:
 		state_machine.start()
 
-	if not state_machine.current_bt_status == Status.RUNNING:
+	if not state_machine.current_status == BTStatus.RUNNING:
 		state_machine.active = false
```
Comment (SirPigeonz) [inline addons/behaviour_toolkit/behaviour_tree/composites/bt_integrated_fsm.gd:13]: Should be `current_bt_status` not `current_status` :)
```suggestion
	if not state_machine.current_bt_status == BTStatus.RUNNING:
```

File: addons/behaviour_toolkit/behaviour_tree/composites/bt_integrated_fsm.gd [hunk 3]
```diff
@@ -5,14 +5,14 @@ class_name BTIntegratedFSM extends BTComposite
...
 	
-	return state_machine.current_bt_status
+	return state_machine.current_status
```
Comment (SirPigeonz) [inline addons/behaviour_toolkit/behaviour_tree/composites/bt_integrated_fsm.gd:16]:
```suggestion
	return state_machine.current_bt_status
```

**Comment by SirPigeonz** [comment]
Ok tested it above bug is the only one I found.
After that one fix I think it's fine to merge. :)

**Comment by ThePat02** [comment]
> Ok tested it and above bug is the only one I found.
After that one fix I think it's fine to merge. :)

If something else is broken pls open another issue. Cant test it right now as I am working in codespaces!

**Pull request merged**